ios/RNSScreenStackHeaderConfig.mm

```diff
@@ -647,8 +697,207 @@ + (void)updateViewController:(UIViewController *)vc
+      return UISemanticContentAttributeForceRightToLeft;
+    case facebook::react::RNSScreenStackHeaderConfigDirection::Ltr:
+      return UISemanticContentAttributeForceLeftToRight;
+  }
+}
+
+- (void)updateProps:(facebook::react::Props::Shared const &)props
+           oldProps:(facebook::react::Props::Shared const &)oldProps
+{
+  [super updateProps:props oldProps:oldProps];
+
+  const auto &oldScreenProps =
+      *std::static_pointer_cast<const facebook::react::RNSScreenStackHeaderConfigProps>(_props);
+  const auto &newScreenProps = *std::static_pointer_cast<const facebook::react::RNSScreenStackHeaderConfigProps>(props);
+
+  BOOL needsNavigationControllerLayout = !_initialPropsSet;
+
+  if (newScreenProps.hidden != !_show) {
+    _show = !newScreenProps.hidden;
+    needsNavigationControllerLayout = YES;
+  }
+
+  if (newScreenProps.translucent != _translucent) {
+    _translucent = newScreenProps.translucent;
+    needsNavigationControllerLayout = YES;
+  }
+
+  _title = RCTNSStringFromStringNilIfEmpty(newScreenProps.title);
+  if (newScreenProps.titleFontFamily != oldScreenProps.titleFontFamily) {
+    _titleFontFamily = RCTNSStringFromStringNilIfEmpty(newScreenProps.titleFontFamily);
+  }
+  _titleFontWeight = RCTNSStringFromStringNilIfEmpty(newScreenProps.titleFontWeight);
+  _titleFontSize = [self getFontSizePropValue:newScreenProps.titleFontSize];
+  _hideShadow = newScreenProps.hideShadow;
+
+  _largeTitle = newScreenProps.largeTitle;
+  if (newScreenProps.largeTitleFontFamily != oldScreenProps.largeTitleFontFamily) {
+    _largeTitleFontFamily = RCTNSStringFromStringNilIfEmpty(newScreenProps.largeTitleFontFamily);
+  }
+  _largeTitleFontWeight = RCTNSStringFromStringNilIfEmpty(newScreenProps.largeTitleFontWeight);
+  _largeTitleFontSize = [self getFontSizePropValue:newScreenProps.largeTitleFontSize];
+  _largeTitleHideShadow = newScreenProps.largeTitleHideShadow;
+
+  _backTitle = RCTNSStringFromStringNilIfEmpty(newScreenProps.backTitle);
+  if (newScreenProps.backTitleFontFamily != oldScreenProps.backTitleFontFamily) {
+    _backTitleFontFamily = RCTNSStringFromStringNilIfEmpty(newScreenProps.backTitleFontFamily);
+  }
+  _backTitleFontSize = [self getFontSizePropValue:newScreenProps.backTitleFontSize];
+  _hideBackButton = newScreenProps.hideBackButton;
+  _disableBackButtonMenu = newScreenProps.disableBackButtonMenu;
+
+  if (newScreenProps.direction != oldScreenProps.direction) {
+    _direction = [self getDirectionPropValue:newScreenProps.direction];
+  }
+
+  // We cannot compare SharedColor because it is shared value.
+  // We could compare color value, but it is more performant to just assign new value
+  _titleColor = RCTUIColorFromSharedColor(newScreenProps.titleColor);
+  _largeTitleColor = RCTUIColorFromSharedColor(newScreenProps.largeTitleColor);
+  _color = RCTUIColorFromSharedColor(newScreenProps.color);
+
+  if (_backgroundSharedColor != newScreenProps.backgroundColor) {
```

Comment:
I don't understand why it is done like this and only with this prop.

Reply:
Good catch. I think it is leftover from first implementation. 
I changed the way to setting this prop according to the comment just above lines you highlighted:
```
  // We cannot compare SharedColor because it is shared value.
  // We could compare color value, but it is more performant to just assign new value
```